kernel/src/fs/epoll/epoll_file.rs

```diff
@@ -460,3 +448,43 @@ impl Observer<IoEvents> for EpollEntry {
         }
     }
 }
+
+impl Drop for EpollEntry {
+    fn drop(&mut self) {
+        let Some(file) = self.file() else {
+            return;
+        };
+        file.unregister_observer(&(self.self_weak() as _)).unwrap();
+    }
+}
+
+struct KeyableEntry(pub Arc<EpollEntry>);
+
+impl PartialOrd for KeyableEntry {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+impl Ord for KeyableEntry {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.0.key.cmp(&other.0.key)
+    }
+}
+impl PartialEq for KeyableEntry {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.key.eq(&other.0.key)
+    }
+}
+impl Eq for KeyableEntry {}
+
+impl Borrow<(FileDesc, KeyableWeak<dyn FileLike>)> for KeyableEntry {
+    fn borrow(&self) -> &(FileDesc, KeyableWeak<dyn FileLike>) {
+        &self.0.key
+    }
+}
```

Comment:
`Borrow` is required by `BTreeSet`. For example, the signature of [`BTreeSet::contains`](https://doc.rust-lang.org/std/collections/struct.BTreeSet.html#method.contains) is:
```rust
pub fn contains<Q>(&self, value: &Q) -> bool
where
    T: Borrow<Q> + Ord,
    Q: Ord + ?Sized,
```

The reason is, also from the [doc](https://doc.rust-lang.org/stable/std/borrow/trait.Borrow.html) of `Borrow` trait:
> In particular Eq, Ord and Hash must be equivalent for borrowed and owned values: x.borrow() == y.borrow() should give the same result as x == y.

If `AsRef` is used, we cannot use the key to look up the `BTreeSet`, since no one guarantees that a comparison between keys has the same semantics as a comparison between values.

Reply:
Thanks for your explanation! Since it's the requirement of BTreeSet, use `Borrow` is suitable.